gunicorn/arbiter.py

```diff
@@ -473,6 +473,8 @@ def manage_workers(self):
         Maintain the number of workers by spawning or killing
         as required.
         """
+        orig_num_workers = self.num_workers
+
         if len(self.WORKERS.keys()) < self.num_workers:
```

Comment:
Oh, wait, I'm wrong. The `waitpid` gives us the worker pid and we pop it with a `None` argument in line 463 so I think it might be harmless to remove it synchronously.